web/src/components/pages/dashboard/challenge/constants.ts

```diff
@@ -26,7 +27,31 @@ export const isBeforeChallenge = (challenge: ChallengeDuration) => {
   return now < challenge.start;
 };
 
-export const pilotDates: ChallengeDuration = {
-  start: new Date('2019-11-18'),
-  end: new Date('2019-12-18'),
+const isValidDate = (dateObj: any) => {
+  return dateObj instanceof Date && !isNaN(dateObj.getTime());
 };
+
+const getChallengeDates = (): ChallengeDuration => {
+  const params = queryString.parse(location.search);
+
+  let defaultDates: ChallengeDuration = {
+    start: new Date('2019-11-18'),
+    end: new Date('2019-12-18'),
+  };
+
+  let challengeDates: ChallengeDuration = defaultDates;
+
+  if (params.challenge_start && isValidDate(new Date(params.challenge_start))) {
+    challengeDates.start = new Date(params.challenge_start);
+  }
+
+  if (params.challenge_end && isValidDate(new Date(params.challenge_end))) {
+    challengeDates.end = new Date(params.challenge_end);
+  }
```

Comment:
Let's discuss this when you're free. My intuition is that having a single `date=` param will allow us to test against the challenge dates easier, and move us closer to being able to test specific weeks.